src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala

```diff
@@ -39,95 +37,31 @@ import scala.util.{ Failure, Success, Try }
   private val datasetMetadataJsonFile = metadataDir / "dataset.json"
 
   /**
-   * @return an information object about the current state of the desposit.
-   */
-  def getStateInfo: Try[StateInfo] = {
-    for {
-      props <- getDepositProps
-      state = State.withName(props.getString("state.label"))
-      description = props.getString("state.description")
-    } yield StateInfo(state, description)
-  }
-
-  /**
-   * Sets changes the state of the deposit. If the state transition is not allow a `Failure` containing
-   * an [[nl.knaw.dans.easy.deposit.Errors.IllegalStateTransitionException]] is returned.
-   *
-   * @param stateInfo the new state
-   * @return Failure or Success(submitId or null)
-   */
-  def setStateInfo(stateInfo: StateInfo): Try[UUID] = {
-    for {
-      props <- checkStateTransition(stateInfo.state)
-      _ = props.setProperty("state.label", stateInfo.state.toString)
-      _ = props.setProperty("state.description", stateInfo.stateDescription.toString)
-      uuid = setSubmitId(stateInfo, props)
-      _ = props.save()
-    } yield uuid
-  }
-
-  private def setSubmitId(stateInfo: StateInfo, props: PropertiesConfiguration): UUID = {
-    // TODO refactor with issue easy-2054 perhaps into some StateManager class
-    //  reduce StateInfo back to the document (alias case class) to exchange with the client
-    val propertyKey = "bag-store.bag-id"
-    if (stateInfo.state == State.submitted) {
-      val uuid = UUID.randomUUID()
-      props.setProperty(propertyKey, uuid)
-      uuid
-    }
-    else { // this must be: rejected -> draft
-      props.clearProperty(propertyKey)
-      null
-    }
-  }
-
-  def checkStateTransition(newState: State): Try[PropertiesConfiguration] = {
-    for {
-      props <- getDepositProps
-      currentState <- getState(props)
-      _ <- if (currentState.canChangeTo(newState)) Success(())
-           else Failure(IllegalStateTransitionException(currentState, newState))
-    } yield props
-  }
-
-  private def getState(props: PropertiesConfiguration): Try[StateInfo.State.Value] = Try {
-    State.withName(props.getString("state.label"))
-  }
-
-  /**
-   * Deletes the deposit.
-   */
-  def delete(): Try[Unit] = for {
-    stateInfo <- getStateInfo
-    _ <- stateInfo.canDelete
-    _ = bagDir.parent.delete()
-  } yield ()
-
-  /**
+   * @param submitBase the base directory with submitted deposits to extract the actual state
    * @return basic information about the deposit.
    */
-  def getDepositInfo: Try[DepositInfo] = {
+  def getDepositInfo(submitBase: File): Try[DepositInfo] = {
     for {
       title <- getDatasetTitle
-      props <- getDepositProps
-      state <- getState(props)
-      created = new DateTime(props.getString("creation.timestamp")).withZone(UTC)
+      stateManager = StateManager(bagDir.parent, submitBase)
+      stateInfo <- stateManager.getStateInfo
+      created = new DateTime(stateManager.draftProps.getString("creation.timestamp")).withZone(UTC)
```

Comment:
why don't you put this in the `stateManager` as well? Like `stateManager.creationTimestamp`

Reply:
I'm abusing the `stateManager` as it happens to have the properties loaded. Otherwise the `creation.timestamp` has nothing to do with the state.